server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -492,17 +493,23 @@ static OverLimitStrategy createOverLimitStrategy(boolean trackRealMemoryUsage) {
         if (trackRealMemoryUsage && jvmInfo.useG1GC().equals("true")
         // messing with GC is "dangerous" so we apply an escape hatch. Not intended to be used.
             && Booleans.parseBoolean(System.getProperty("es.real_memory_circuit_breaker.g1_over_limit_strategy.enabled"), true)) {
-            TimeValue lockTimeout = TimeValue.timeValueMillis(
-                Integer.parseInt(System.getProperty("es.real_memory_circuit_breaker.g1_over_limit_strategy.lock_timeout_ms", "500"))
+
+            long lockTimeoutInMillis = Integer.parseInt(
+                System.getProperty("es.real_memory_circuit_breaker.g1_over_limit_strategy.lock_timeout_ms", "500")
             );
+            TimeValue lockTimeout = TimeValue.timeValueMillis(lockTimeoutInMillis);
+            // TODO: consider a separate property for the lock timeout under full GC
+            TimeValue fullGCLockTimeout = TimeValue.timeValueMillis(lockTimeoutInMillis * 2);
             // hardcode interval, do not want any tuning of it outside code changes.
             return new G1OverLimitStrategy(
                 jvmInfo,
                 HierarchyCircuitBreakerService::realMemoryUsage,
                 createYoungGcCountSupplier(),
                 System::currentTimeMillis,
                 500,
-                lockTimeout
+                5000,
```

Comment:
++ 5 seconds seems reasonable here (for fullGCMinimumInterval )

Reply:
Yes, that was what I remember being discussed on slack. I'd like to try a couple of different values though